crates/codegen/runtime/generator/src/kinds.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::BTreeSet;
+
+use codegen_language_definition::model::{self, Identifier, Item};
+use serde::Serialize;
+
+#[derive(Default, Serialize)]
+pub struct KindsModel {
```

Comment:
I think we might be overloading the term "kinds" here. While it is accurate for `TerminalKind` and `NonTerminalKind`, it is no longer the case for labels, and it was never the case for things like lexical contexts. I'll try to think of suggestions here. Just TOL. Not blocking of course.

Reply:
The lexical contexts enum as-is was an afterthought, I agree, however the labels (prev. edge kinds) are better found next to the other kinds that define the syntax tree, i.e. terminals and non-terminals, especially since they're used together (defined in the same place for the metaslang cst, used in queries etc.).

I plan separating the lexical bits into a dedicated generated lexer module later for clarity, so it'd be great if we could keep it here for now 🙏

Reply:
Sounds good.